MTA-4887: Ensure that JDK requirements for Linux indicate to install Development package (#129)
migtools/mta-documentation: docs/topics/installing-cli-tool.adoc

[IMPORTANT]
====
Starting from {ProductShortName} 7.2.0, you can run the application analysis for Java applications in the containerless mode.

The containerless mode is set by default and is used automatically if all requirements are met. For more information, see xref:running-the-containerless-mta-cli_cli-guide[Running the containerless CLI].

However, if you want to analyze applications in languages other than Java or, for example, use xref:mta-cli-transform_cli-guide[transformation commands], you still need to use containers.
====

.Prerequisites for containerized CLI

The following are the prerequisites for the {ProductFullName} CLI installation (container):

* 8 GB RAM is required for the {ProductShortName} {CLIName}.

.Prerequisites for containerless CLI

The following are the prerequisites if you want to to run {ProductShortName} {CLIName} as a xref:running-the-containerless-mta-cli_cli-guide[containerless CLI] mode:

* Java Development Kit (JDK) is installed. {ProductShortName} supports the following JDKs:

** OpenJDK 17 or later
** Oracle JDK 17 or later
** Eclipse Temurin™ JDK 17 or later

* You have link:https://maven.apache.org/guides/[Maven] configured.

> ////

Author: ```suggestion

////
```

[IMPORTANT]
====
Starting from {ProductShortName} 7.2.0, you can run the application analysis for Java applications in the containerless mode.

The containerless mode is set by default and is used automatically if all requirements are met. For more information, see xref:running-the-containerless-mta-cli_cli-guide[Running the containerless CLI].

However, if you want to analyze applications in languages other than Java or, for example, use xref:mta-cli-transform_cli-guide[transformation commands], you still need to use containers.
====

.Prerequisites for containerized CLI

The following are the prerequisites for the {ProductFullName} CLI installation (container):

* 8 GB RAM is required for the {ProductShortName} {CLIName}.

.Prerequisites for containerless CLI

The following are the prerequisites if you want to to run {ProductShortName} {CLIName} as a xref:running-the-containerless-mta-cli_cli-guide[containerless CLI] mode:

* Java Development Kit (JDK) is installed. {ProductShortName} supports the following JDKs:

** OpenJDK 17 or later
** Oracle JDK 17 or later
** Eclipse Temurin™ JDK 17 or later

> * You have link:https://maven.apache.org/guides/[Maven] configured.

Author: ```suggestion
* You have link:https://maven.apache.org/guides/[Maven] installed and configured. For more information, see link:https://maven.apache.org/install.html[Apache Maven Installation].
```

////
** To run the {MavenName} on OpenJDK 17 or Oracle JDK17, you first need to set MAVEN_OPTS on the command line by running the following command:
+
[source,terminal,subs="attributes+"]
----
export MAVEN_OPTS="--add-modules=java.se --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.util.stream=ALL-UNNAMED"
----
////

* You have the `JAVA_HOME` environmental variable set.

* Linux installation requires:

> ** A full Java JDK installed, including a complier provided by the `devel` package, as having only the Java Runtime (JRE) installed causes issues during analysis. For more information about installing Red Hat build of OpenJDK 17 on {op-system-base-full}, see link:https://docs.redhat.com/en/documentation/red_hat_build_of_openjdk/17/html-single/installing_and_using_red_hat_build_of_openjdk_17_on_rhel/index#installing-jre-on-rhel-using-archive_openjdk[Installing Red Hat build of OpenJDK on RHEL by using yum].

Reviewer: Typo in "complier". The `devel` package is only relevant for Fedora.

> ** A full Java JDK installed, including a complier provided by the `devel` package, as having only the Java Runtime (JRE) installed causes issues during analysis. For more information about installing Red Hat build of OpenJDK 17 on {op-system-base-full}, see link:https://docs.redhat.com/en/documentation/red_hat_build_of_openjdk/17/html-single/installing_and_using_red_hat_build_of_openjdk_17_on_rhel/index#installing-jre-on-rhel-using-archive_openjdk[Installing Red Hat build of OpenJDK on RHEL by using yum].

Author: ```suggestion
** A full Java JDK installed, including a compiler provided by the `devel` package, as having only the Java Runtime (JRE) installed causes issues during analysis. For more information about installing Red Hat build of OpenJDK 17 on {op-system-base-full}, see link:https://docs.redhat.com/en/documentation/red_hat_build_of_openjdk/17/html-single/installing_and_using_red_hat_build_of_openjdk_17_on_rhel/index#installing-jre-on-rhel-using-archive_openjdk[Installing Red Hat build of OpenJDK on RHEL by using yum].
```

* macOS installation requires:

> ** A full Java JDK installed, including a complier provided by the `devel` package, as having only the Java Runtime (JRE) installed causes issues during analysis.

Reviewer: Typo in "complier"

.Prerequisites

The following are the prerequisites for the {ProductFullName} CLI installation:

* Java Development Kit (JDK) is installed. {ProductShortName} supports the following JDKs:

** OpenJDK 17 or later
** Oracle JDK 17 or later
** Eclipse Temurin™ JDK 17 or later

* 8 GB RAM is required for the MTA CLI.

// Let us flag to MTA-QE channel
// * macOS installation: the value of `maxproc` must be `2048` or greater.

* Linux installation requires the `java-<JDK-version>-devel` package and not only the `java--<JDK-version>` package.
+
For example, to install OpenJDK 17 on {op-system-base-full}, you would run:
+
[source,terminal]
----
sudo dnf install java-17-openjdk java-17-openjdk-devel
----

> // Igor do we need to keep this bulletpoint

Author: @ibragins - does this bulletpoint about *ed Hat Container Registry Authentication* need to kept as it mentioned later in the doc?

Reviewer 2: _Red Hat Container Registry Authentication_ needs to be kept for pulling container images so that container analysis (all non-java, legacy java, tranformation, etc) will work

> * Red Hat Container Registry Authentication for `registry.redhat.io`. Red Hat distributes container images from `registry.redhat.io`, which requires authentication. For more details, see link:https://access.redhat.com/RegistryAuthentication[Red Hat Container Registry Authentication].

Author: ```suggestion
* You are required to have a valid username and password for Red Hat Container Registry Authentication, for `registry.redhat.io`, if you are going to use the {ProductShortName} {CLIName}. 
+
Red Hat distributes container images from `registry.redhat.io`, which requires authentication. For more details, see link:https://access.redhat.com/RegistryAuthentication[Red Hat Container Registry Authentication].
```

.Prerequisites

The following are the prerequisites for the {ProductFullName} CLI installation:

* Java Development Kit (JDK) is installed. {ProductShortName} supports the following JDKs:

** OpenJDK 17 or later
** Oracle JDK 17 or later

> ** Eclipse Temurin™ JDK 17 or later

Reviewer 2: This is required if user wants to run in **local** mode and is not required if user will run in container mode (legacy java or any other language)

* 8 GB RAM is required for the {ProductShortName} {CLIName}.

Linux installation requires a full Java JDK installed, as having only the Java Runtime (JRE) installed causes issues during analysis.

* To install the {ProductShortName} {CLIName}, you must have a valid username and password for Red Hat Container Registry Authentication, `registry.redhat.io`. 
+

> Red Hat distributes container images from `registry.redhat.io`, which requires authentication. For more details, see link:https://access.redhat.com/RegistryAuthentication[Red Hat Container Registry Authentication].

Reviewer 2: This is required if user wants to use a container runtime, which is required if user wants to run java analysis in legacy mode or any other analysis. It is not required for running in **local** mode

Reviewer 2: One more thing, running in **local** mode alse requires Mavan and $JAVA_HOME configured properly, relevant for EACH platform

.Prerequisites

The following are the prerequisites for the {ProductFullName} CLI installation:

* Java Development Kit (JDK) is installed. {ProductShortName} supports the following JDKs:

** OpenJDK 17 or later
** Oracle JDK 17 or later
** Eclipse Temurin™ JDK 17 or later

* 8 GB RAM is required for the {ProductShortName} {CLIName}.

// Let us flag to MTA-QE channel

> // * macOS installation: the value of `maxproc` must be `2048` or greater.

Reviewer 2: I don't see any reason to keep it there @anarnold97

> // Let us flag to MTA-QE channel
> // * macOS installation: the value of `maxproc` must be `2048` or greater.

Author: ```suggestion
```

> * Linux installation requires a full Java JDK installed, including the `java-<JDK-version>-devel` package, as having only the Java Runtime (JRE) installed causes issues during analysis.

Author: ```suggestion
Linux installation requires a full Java JDK installed, as having only the Java Runtime (JRE) installed causes issues during analysis.
```

.Prerequisites

The following are the prerequisites for the {ProductFullName} CLI installation:

* Java Development Kit (JDK) is installed. {ProductShortName} supports the following JDKs:

** OpenJDK 17 or later
** Oracle JDK 17 or later
** Eclipse Temurin™ JDK 17 or later

> * 8 GB RAM is required for the MTA CLI.

Author: ```suggestion
* 8 GB RAM is required for the {ProductShortName} {CLIName}.
```

// Let us flag to MTA-QE channel
// * macOS installation: the value of `maxproc` must be `2048` or greater.

> * Linux installation requires the `java-<JDK-version>-devel` package and not only the `java--<JDK-version>` package.

Author: ```suggestion
* Linux installation requires a full Java JDK installed, including the `java-<JDK-version>-devel` package, as having only the Java Runtime (JRE) installed causes issues during analysis.
```

+
For example, to install OpenJDK 17 on {op-system-base-full}, you would run:
+
[source,terminal]
----
sudo dnf install java-17-openjdk java-17-openjdk-devel

> ----

Reviewer: @anarnold97 regarding this whole "Linux installation" part above, I think I would remove it entirely. It should be known to the developer how to properly install a JDK distribution. The definition of "JDK" itself implies having all the development tools.

What happened to @ibragins was that he had installed only the "JRE" (Java Runtime Environment), which in the particular case of the Fedora package manager (not all Linux distros) is called `java-17-openjdk`.

If you don't want to remove it entirely, I would maybe switch it to an asterisk saying something like
> * Please note that MTA requires a fully installed JDK. Having a JRE only will cause issues during analysis.
and leave it to the developer to figure out how to install it for his particular OS.

Reviewer 2: This is not correct @jmle , I installed OpenJDK, not JRE. The fact is that running `sudo dnf install java-17-openjdk` on Fedora doesn't install a devel package which actually caused the problem and is the reason of this conversation

> +
> For example, to install OpenJDK 17 on {op-system-base-full}, you would run:
> +
> [source,terminal]
> ----
> sudo dnf install java-17-openjdk java-17-openjdk-devel
> ----

Author: ```suggestion
```

// Igor do we need to keep this bulletpoint

* To install the {ProductShortName} {CLIName}, you must have a valid username and password for Red Hat Container Registry Authentication, `registry.redhat.io`. 
+

> Red Hat distributes container images from `registry.redhat.io`, which requires authentication. For more details, see link:https://access.redhat.com/RegistryAuthentication[Red Hat Container Registry Authentication].

Author: @ibragins - is this what you meant? sorry was not sure
